src/world_interface/motor/base_motor.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#include "../../control/JacobianVelController.h"
+#include "../../utils/scheduler.h"
+#include "../data.h"
+
+/**
+ * @namespace robot
+ * @brief Collection of functions for manipulating a motor.
+ */
+namespace robot {
+/**
+ * An abstract motor class
+ *
+ */
+class base_motor {
+public:
+	/**
+	 * @brief Constructor for base motor.
+	 *
+	 * @param motor The motor id to manipulate.
+	 * @param hasPosSensor Boolean to indicate if the motor has a position sensor.
+	 */
+	base_motor(robot::types::motorid_t motor, bool hasPosSensor);
+
```

Comment:
Copying motors should be banned, so we should delete the copy constructor and operator:

```
base_motor(const base_motor&) = delete;

base_motor& operator=(const base_motor&) = delete;
```